rqt_gui_py/src/rqt_gui_py/ros_py_plugin_provider.py

```diff
@@ -29,77 +29,32 @@
-from qt_gui.errors import PluginLoadError
-
-from python_qt_binding.QtCore import qDebug, Qt, qWarning, Signal
-from python_qt_binding.QtWidgets import QMessageBox
 
+import rclpy
 from rqt_gui.rospkg_plugin_provider import RospkgPluginProvider
 
 
 class RosPyPluginProvider(CompositePluginProvider):
 
-    _master_found_signal = Signal(int)
-
     def __init__(self):
         super(RosPyPluginProvider, self).__init__(
             [RospkgPluginProvider('rqt_gui', 'rqt_gui_py::Plugin')])
         self.setObjectName('RosPyPluginProvider')
         self._node_initialized = False
-        self._wait_for_master_dialog = None
-        self._wait_for_master_thread = None
+        self._node = None
 
     def load(self, plugin_id, plugin_context):
-        self._check_for_master()
         self._init_node()
         return super(RosPyPluginProvider, self).load(plugin_id, plugin_context)
 
-    def _check_for_master(self):
-        # check if master is available
-        try:
-            rospy.get_master().getSystemState()
-            return
-        except Exception:
-            pass
-        # spawn thread to detect when master becomes available
-        self._wait_for_master_thread = threading.Thread(target=self._wait_for_master)
-        self._wait_for_master_thread.start()
-        self._wait_for_master_dialog = QMessageBox(QMessageBox.Question,
-                                                   self.tr('Waiting for ROS master'),
-                                                   self.tr("Could not find ROS master. Either start a 'roscore' or "
-                                                           "abort loading the plugin."),
-                                                   QMessageBox.Abort)
-        self._master_found_signal.connect(self._wait_for_master_dialog.done, Qt.QueuedConnection)
-        button = self._wait_for_master_dialog.exec_()
-        # check if master existence was not detected by background thread
-        no_master = button != QMessageBox.Ok
-        self._wait_for_master_dialog.deleteLater()
-        self._wait_for_master_dialog = None
-        if no_master:
-            raise PluginLoadError('RosPyPluginProvider._init_node() could not find ROS master')
-
-    def _wait_for_master(self):
-        while True:
-            time.sleep(0.1)
-            if not self._wait_for_master_dialog:
-                break
-            try:
-                rospy.get_master().getSystemState()
-            except Exception:
-                continue
-            self._master_found_signal.emit(QMessageBox.Ok)
-            break
-
     def _init_node(self):
         # initialize node once
         if not self._node_initialized:
             name = 'rqt_gui_py_node_%d' % os.getpid()
             qDebug('RosPyPluginProvider._init_node() initialize ROS node "%s"' % name)
-            rospy.init_node(name, disable_signals=True)
+            rclpy.init()
+            self._node = rclpy.create_node(name)
```

Comment:
Is this called automagically or does the user need to call _init_node? 

Also, you may want to make explicit destroying the node an rclpy.shutdown() in a __del__(self) method

Reply:
_init_node() is called from the public method load() above. I can't actually find a plugin yet that implements this class.

I'll add similar unload and _destroy_node methods.